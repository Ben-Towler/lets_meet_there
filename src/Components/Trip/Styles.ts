import styled from 'styled-components';

export const StyledTrip = styled.section`
`;
