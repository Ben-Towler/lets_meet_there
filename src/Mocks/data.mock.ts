export const placeMocks = {
    65698: {
      CityId: 'LOND',
      CityName: 'London',
      CountryName: 'United Kingdom',
      IataCode: 'LHR',
      Name: 'London Heathrow',
      PlaceId: 65698,
      SkyscannerCode: 'LHR',
      Type: 'Station'
    }
  };

export const quotes = [{
  QuoteId: 1,
  MinPrice: 349,
  Direct: false,
  OutboundLeg: {
    CarrierIds: [
      1090
    ],
    OriginId: 65698,
    DestinationId: 65698,
    DepartureDate: '2020-06-26T00:00:00'
  },
  InboundLeg: {
    CarrierIds: [
      1755
    ],
    OriginId: 65698,
    DestinationId: 65698,
    DepartureDate: '2020-06-29T00:00:00'
  },
  QuoteDateTime: '2020-06-17T21:12:00'
}];

export const carriers = [{

}];

export const flightOne = {
  Direct: false,
  InboundLeg: {
    CarrierIds: [1324],
    DepartureDate: '2020-06-27T00:00:00',
    DestinationId: 65698,
    OriginId: 65698,
  },
  MinPrice: 213,
  OutboundLeg: {
    CarrierIds: [1324],
    DepartureDate: '2020-06-25T00:00:00',
    DestinationId: 65698,
    OriginId: 65698,
  },
  QuoteDateTime: '2020-06-17T23:03:00',
  QuoteId: 19
}

export const flightTwo = {
  Direct: false,
  InboundLeg: {
    CarrierIds: [1324],
    DepartureDate: '2020-06-27T00:00:00',
    DestinationId: 65698,
    OriginId: 65698
  },
  MinPrice: 192,
  OutboundLeg: {
    CarrierIds: [1324],
    DepartureDate: '2020-06-25T00:00:00',
    DestinationId: 65698,
    OriginId: 65698
  },
  QuoteDateTime: '2020-06-15T19:17:00',
  QuoteId: 10
}

export const flightListMock = [[
  66328,
  [
    {
      QuoteId: 59,
      MinPrice: 177,
      Direct: false,
      OutboundLeg: {
        CarrierIds: [
          1090
        ],
        OriginId: 65698,
        DestinationId: 65698,
        DepartureDate: '2020-06-26T00:00:00'
      },
      InboundLeg: {
        CarrierIds: [
          1090
        ],
        OriginId: 65698,
        DestinationId: 65698,
        DepartureDate: '2020-06-29T00:00:00'
      },
      QuoteDateTime: '2020-06-18T10:10:00'
    }
  ], [
    {
      QuoteId: 9,
      MinPrice: 26,
      Direct: true,
      OutboundLeg: {
        CarrierIds: [
          1090
        ],
        OriginId: 65698,
        DestinationId: 65698,
        DepartureDate: '2020-06-26T00:00:00'
      },
      InboundLeg: {
        CarrierIds: [
          1090
        ],
        OriginId: 65698,
        DestinationId: 65698,
        DepartureDate: '2020-06-29T00:00:00'
      },
      QuoteDateTime: '2020-06-18T16:38:00'
    }
  ]
]];

export const favLocation = {
  city: 'Berlin',
  country: 'Germany'
}

export const userCity = 'Berlin';
export const friendCity = 'Berlin';

export const searchDetailsForRemoveHandler = {
  origin: '',
  destination: '',
  outboundDate: '',
  inboundDate: ''
}