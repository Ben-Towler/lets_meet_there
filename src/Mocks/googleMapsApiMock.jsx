export const setupGoogleMock = () => {
  // Mock Google Maps JavaScript API 
  const google = {
    maps: {
      places: {
        Autocomplete: class {},
        AutocompleteService:class{},
        PlacesServiceStatus: {
          INVALID_REQUEST: 'INVALID_REQUEST',
          NOT_FOUND: 'NOT_FOUND',
          OK: 'OK',
          OVER_QUERY_LIMIT: 'OVER_QUERY_LIMIT',
          REQUEST_DENIED: 'REQUEST_DENIED',
          UNKNOWN_ERROR: 'UNKNOWN_ERROR',
          ZERO_RESULTS: 'ZERO_RESULTS',
        },
      },
      Geocoder:class{},
      GeocoderStatus: {
        ERROR: 'ERROR',
        INVALID_REQUEST: 'INVALID_REQUEST',
        OK: 'OK',
        OVER_QUERY_LIMIT: 'OVER_QUERY_LIMIT',
        REQUEST_DENIED: 'REQUEST_DENIED',
        UNKNOWN_ERROR: 'UNKNOWN_ERROR',
        ZERO_RESULTS: 'ZERO_RESULTS',
      },
    },
  };
  global.window.google = google;
};

