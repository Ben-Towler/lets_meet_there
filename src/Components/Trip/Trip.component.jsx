import React from 'react';
import { useSelector } from 'react-redux';
import { StyledTrip } from './Styles.js';
import { Flight, FlightSummary } from 'Components';
import { firestore } from '../../Services/firebase.utils';

const Trip = ({ yourFlight, friendsFlight, location, favourites, userCity, friendCity, favLocation, removeFromFavouritesHandler, searchDetailsForRemoveHandler, user }) => {
  const places = useSelector((state) => state.places);
  const carriers = useSelector((state) => state.carriers);

  const meetingLocation = {
    city: favLocation ? favLocation.city : places[location].CityName,
    country: favLocation ? favLocation.country : places[location].CountryName
  };

  const addToFavouritesHandler = async (e) => {
    const requestData = {
      userRequest: {
        origin: places[yourFlight.OutboundLeg.OriginId].CityId,
        destination: places[location].CityId,
        outboundDate: yourFlight.OutboundLeg.DepartureDate.slice(0, 10),
        inboundDate: yourFlight.InboundLeg.DepartureDate.slice(0, 10)
      },
      friendRequest: {
        origin: places[friendsFlight.OutboundLeg.OriginId].CityId,
        destination: places[location].CityId,
        outboundDate: friendsFlight.OutboundLeg.DepartureDate.slice(0, 10),
        inboundDate: friendsFlight.InboundLeg.DepartureDate.slice(0, 10)
      }
    }
    const userRef = await firestore.doc(`users/${user.id}`);
    userRef.update({ favourites: JSON.stringify([...user.favourites, requestData]) });
  }

  const favouritesButton = !favourites ? <button onClick={addToFavouritesHandler}>Add To Favourites</button> :
    <button onClick={() => removeFromFavouritesHandler (
      searchDetailsForRemoveHandler.origin, 
      searchDetailsForRemoveHandler.destination, 
      searchDetailsForRemoveHandler.outboundDate, 
      searchDetailsForRemoveHandler.inboundDate
    )}>Remove From Favourites</button>
  ;

  const yourFlightPlaces = {
    inbound: {
      origin: favLocation ? favLocation.city : places[yourFlight.InboundLeg.OriginId],
      destination: userCity ? userCity : places[yourFlight.InboundLeg.DestinationId]
    },
    outbound: {
      origin: userCity ? userCity : places[yourFlight.OutboundLeg.OriginId],
      destination: favLocation ? favLocation.city : places[yourFlight.OutboundLeg.DestinationId]
    }
  }

  const friendsFlightPlaces = {
    inbound: {
      origin: favLocation ? favLocation.city :  places[friendsFlight.InboundLeg.OriginId],
      destination: friendCity ? friendCity : places[friendsFlight.InboundLeg.DestinationId]
    },
    outbound: {
      origin: friendCity ? friendCity : places[friendsFlight.OutboundLeg.OriginId],
      destination: favLocation ? favLocation.city :  places[friendsFlight.OutboundLeg.DestinationId]
    }
  }

  return (
    <StyledTrip>

      <Flight 
        flight={ yourFlight } 
        city={ userCity && userCity }
        favLocation={ favLocation && favLocation }
        places={ yourFlightPlaces }
      />
      
      <FlightSummary 
        location={ meetingLocation } 
        favLocation={ favLocation && favLocation }
      />

      {user ? favouritesButton : null }

      <Flight 
        flight={ friendsFlight }
        favLocation={ favLocation && favLocation }
        places={ friendsFlightPlaces }
        city={ friendCity && friendCity }
      />

    </StyledTrip>
  );
};

export default Trip;
