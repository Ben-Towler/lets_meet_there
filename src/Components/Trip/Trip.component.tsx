import React from 'react';
import { useSelector } from 'react-redux';
import { StyledTrip } from './Styles';
import { Flight, FlightSummary, Button } from 'Components';
import { firestore } from '../../Services/firebase.utils';
import {Flight as FlightInterface} from 'Interfaces/Flight';
import {Location} from 'Interfaces/Location';
import {User as UserInterface} from 'Interfaces/User';


interface Props {
  yourFlight: FlightInterface;
  friendsFlight: FlightInterface;
  location?: number;
  favourites?: Boolean;
  userCity?: string;
  friendCity?: string;
  favLocation?: Location;
  removeFromFavouritesHandler?: Function
  searchDetailsForRemoveHandler?: {origin: string, destination: string, outboundDate: string, inboundDate: string};
  user?: UserInterface;
}

// interface StatePlace {
//   Name: string, 
//   PlaceId: number, 
//   SkyscannerCode: string, 
//   type: string
// }

// {[key: any]: StatePlace}

function Trip ({ yourFlight, friendsFlight, location, favourites, userCity, friendCity, favLocation, removeFromFavouritesHandler, searchDetailsForRemoveHandler, user }: Props): JSX.Element {
  const places = useSelector((state: any) => state.places);
  const carriers = useSelector((state: any) => state.carriers);

  const meetingLocation = {
    city: favLocation ? favLocation.city : places[location!].CityName,
    country: favLocation ? favLocation.country : places[location!].CountryName
  };

  const addToFavouritesHandler = async () => {
    const requestData = {
      userRequest: {
        origin: places[yourFlight.OutboundLeg.OriginId].CityId,
        destination: places[location!].CityId,
        outboundDate: yourFlight.OutboundLeg.DepartureDate.slice(0, 10),
        inboundDate: yourFlight.InboundLeg.DepartureDate.slice(0, 10)
      },
      friendRequest: {
        origin: places[friendsFlight.OutboundLeg.OriginId].CityId,
        destination: places[location!].CityId,
        outboundDate: friendsFlight.OutboundLeg.DepartureDate.slice(0, 10),
        inboundDate: friendsFlight.InboundLeg.DepartureDate.slice(0, 10)
      }
    }
    if (user){
      const userRef = await firestore.doc(`users/${user.id}`);
      userRef.update({ favourites: JSON.stringify([...user.favourites, requestData]) });
    }
  }

  const favouritesButton = !favourites ? <button onClick={addToFavouritesHandler}>Add To Favourites</button> :
    removeFromFavouritesHandler && searchDetailsForRemoveHandler && 
    <button onClick={() => removeFromFavouritesHandler (
      searchDetailsForRemoveHandler.origin, 
      searchDetailsForRemoveHandler.destination, 
      searchDetailsForRemoveHandler.outboundDate, 
      searchDetailsForRemoveHandler.inboundDate
    )}>Remove From Favourites</button>

  ;

  const yourFlightPlaces = {
    inbound: {
      origin: favLocation ? favLocation.city : places[yourFlight.InboundLeg.OriginId],
      destination: userCity ? userCity : places[yourFlight.InboundLeg.DestinationId]
    },
    outbound: {
      origin: userCity ? userCity : places[yourFlight.OutboundLeg.OriginId],
      destination: favLocation ? favLocation.city : places[yourFlight.OutboundLeg.DestinationId]
    }
  }

  const friendsFlightPlaces = {
    inbound: {
      origin: favLocation ? favLocation.city :  places[friendsFlight.InboundLeg.OriginId],
      destination: friendCity ? friendCity : places[friendsFlight.InboundLeg.DestinationId]
    },
    outbound: {
      origin: friendCity ? friendCity : places[friendsFlight.OutboundLeg.OriginId],
      destination: favLocation ? favLocation.city :  places[friendsFlight.OutboundLeg.DestinationId]
    }
  }

  return (
    <StyledTrip>

      <Flight 
        flight={ yourFlight } 
        city={ userCity && userCity }
        favLocation={ favLocation && favLocation }
        places={ yourFlightPlaces }
      />
      
      <FlightSummary 
        location={ meetingLocation } 
      />

      {user ? favouritesButton : null }

      <Flight 
        flight={ friendsFlight }
        favLocation={ favLocation && favLocation }
        places={ friendsFlightPlaces }
        city={ friendCity && friendCity }
      />

    </StyledTrip>
  );
};

export default Trip;
