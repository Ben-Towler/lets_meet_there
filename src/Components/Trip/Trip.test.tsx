import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store'
import Trip from './Trip.component';
import Adapter from 'enzyme-adapter-react-16';
import * as ReactReduxHooks from '../../Mocks/hooks';
import { configure } from 'enzyme';
import thunk from 'redux-thunk';
import {placeMocks, quotes, carriers, flightOne, flightTwo, favLocation, userCity, friendCity, searchDetailsForRemoveHandler, userMockNoData} from '../../Mocks/data.mock';
configure({adapter: new Adapter()});

describe('Trip', () => {
  let store = configureStore([thunk])({
    places: placeMocks,
    quotes: quotes,
    carriers: carriers
  });
  let wrapper: any;
  
  beforeEach(() => {
    store = configureStore([thunk])({
      places: placeMocks,
      quotes: quotes,
      carriers: carriers
    });

    jest.mock('react-redux', () => ({
      useSelector: jest.fn()
    }));

    jest
    .spyOn(ReactReduxHooks, 'useSelector')
    .mockImplementation(state => store.getState());

    wrapper = render (
      <Provider store={store}>
        <Trip 
          yourFlight={flightOne} 
          friendsFlight={flightTwo}
          location={65698}
        />
      </Provider>
    )
  });
  
  test('Should render inner DOM components', () => {
    // wrapper.findAllByText('Book').then(data => data);
    
  });

  test('snapshot matches', () => {
    const result = render(
      <Provider store={store}>
        <Trip 
          yourFlight={flightOne} 
          friendsFlight={flightTwo}
          location={65698}
        />
      </Provider>
    )
    expect(result).toMatchSnapshot()
  });

  test('snapshot matches', () => {
    const result = render(
      <Provider store={store}>
        <Trip 
          yourFlight={flightOne} 
          friendsFlight={flightTwo}
          location={65698}
          favourites={true}
          userCity={userCity}
          friendCity={friendCity}
          favLocation={favLocation}
          searchDetailsForRemoveHandler={searchDetailsForRemoveHandler}
        />
      </Provider>
    )
    expect(result).toMatchSnapshot()
  });

  test('should render and click', () => {
    const mockFunction = jest.fn();
    const {getByText} = render(  
      <Provider store={store}>
        <Trip 
          yourFlight={flightOne} 
          friendsFlight={flightTwo}
          location={65698}
          user={userMockNoData}
          
        />
      </Provider>)

    const btn = getByText('Add To Favourites');
    btn.onclick = mockFunction;
    btn.click()
    expect(mockFunction).toHaveBeenCalled();
    expect(btn).toBeInTheDocument();
  })

  test('should render and click', () => {
    const mockFunction = jest.fn();
    const {getByText} = render(  
      <Provider store={store}>
        <Trip 
          yourFlight={flightOne} 
          friendsFlight={flightTwo}
          location={65698}
          favourites={true}
          userCity={userCity}
          friendCity={friendCity}
          favLocation={favLocation}
          user={userMockNoData}
          removeFromFavouritesHandler={mockFunction}
          searchDetailsForRemoveHandler={searchDetailsForRemoveHandler}
        />
      </Provider>)

    const btn = getByText('Remove From Favourites');
    btn.onclick = mockFunction;
    btn.click()
    expect(mockFunction).toHaveBeenCalled();
    expect(btn).toBeInTheDocument();
  })
});