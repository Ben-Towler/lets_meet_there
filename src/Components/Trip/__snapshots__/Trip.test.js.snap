// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Trip snapshot matches 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="sc-AxjAm fDEmJb"
      >
        <div
          class="sc-AxheI dVkpdt"
        >
          London Heathrow-&gt;
          London Heathrow
          London Heathrow-&gt;
          London Heathrow
          InDirect
          213
          <a
            aria-label="Book Flights"
            class="sc-AxmLO dTodBP"
            href="https://www.skyscanner.net/transport/flights/Lond/Lond/2020-06-25/2020-06-27"
            rel="noopener noreferrer"
            target="_blank"
            title="Book Flights"
          >
            Book Your Flights
          </a>
        </div>
        <div
          class="sc-Axmtr kkOJu"
        >
          London
          United Kingdom
        </div>
        <div
          class="sc-AxheI dVkpdt"
        >
          London Heathrow-&gt;
          London Heathrow
          London Heathrow-&gt;
          London Heathrow
          InDirect
          192
          <a
            aria-label="Book Flights"
            class="sc-AxmLO dTodBP"
            href="https://www.skyscanner.net/transport/flights/Lond/Lond/2020-06-25/2020-06-27"
            rel="noopener noreferrer"
            target="_blank"
            title="Book Flights"
          >
            Book Your Flights
          </a>
        </div>
      </section>
    </div>
    <div>
      <section
        class="sc-AxjAm fDEmJb"
      >
        <div
          class="sc-AxheI dVkpdt"
        >
          London Heathrow-&gt;
          London Heathrow
          London Heathrow-&gt;
          London Heathrow
          InDirect
          213
          <a
            aria-label="Book Flights"
            class="sc-AxmLO dTodBP"
            href="https://www.skyscanner.net/transport/flights/Lond/Lond/2020-06-25/2020-06-27"
            rel="noopener noreferrer"
            target="_blank"
            title="Book Flights"
          >
            Book Your Flights
          </a>
        </div>
        <div
          class="sc-Axmtr kkOJu"
        >
          London
          United Kingdom
        </div>
        <div
          class="sc-AxheI dVkpdt"
        >
          London Heathrow-&gt;
          London Heathrow
          London Heathrow-&gt;
          London Heathrow
          InDirect
          192
          <a
            aria-label="Book Flights"
            class="sc-AxmLO dTodBP"
            href="https://www.skyscanner.net/transport/flights/Lond/Lond/2020-06-25/2020-06-27"
            rel="noopener noreferrer"
            target="_blank"
            title="Book Flights"
          >
            Book Your Flights
          </a>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="sc-AxjAm fDEmJb"
    >
      <div
        class="sc-AxheI dVkpdt"
      >
        London Heathrow-&gt;
        London Heathrow
        London Heathrow-&gt;
        London Heathrow
        InDirect
        213
        <a
          aria-label="Book Flights"
          class="sc-AxmLO dTodBP"
          href="https://www.skyscanner.net/transport/flights/Lond/Lond/2020-06-25/2020-06-27"
          rel="noopener noreferrer"
          target="_blank"
          title="Book Flights"
        >
          Book Your Flights
        </a>
      </div>
      <div
        class="sc-Axmtr kkOJu"
      >
        London
        United Kingdom
      </div>
      <div
        class="sc-AxheI dVkpdt"
      >
        London Heathrow-&gt;
        London Heathrow
        London Heathrow-&gt;
        London Heathrow
        InDirect
        192
        <a
          aria-label="Book Flights"
          class="sc-AxmLO dTodBP"
          href="https://www.skyscanner.net/transport/flights/Lond/Lond/2020-06-25/2020-06-27"
          rel="noopener noreferrer"
          target="_blank"
          title="Book Flights"
        >
          Book Your Flights
        </a>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Trip snapshot matches 2`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="sc-AxjAm fDEmJb"
      >
        <div
          class="sc-AxheI dVkpdt"
        >
          London Heathrow-&gt;
          London Heathrow
          London Heathrow-&gt;
          London Heathrow
          InDirect
          213
          <a
            aria-label="Book Flights"
            class="sc-AxmLO dTodBP"
            href="https://www.skyscanner.net/transport/flights/Lond/Lond/2020-06-25/2020-06-27"
            rel="noopener noreferrer"
            target="_blank"
            title="Book Flights"
          >
            Book Your Flights
          </a>
        </div>
        <div
          class="sc-Axmtr kkOJu"
        >
          London
          United Kingdom
        </div>
        <div
          class="sc-AxheI dVkpdt"
        >
          London Heathrow-&gt;
          London Heathrow
          London Heathrow-&gt;
          London Heathrow
          InDirect
          192
          <a
            aria-label="Book Flights"
            class="sc-AxmLO dTodBP"
            href="https://www.skyscanner.net/transport/flights/Lond/Lond/2020-06-25/2020-06-27"
            rel="noopener noreferrer"
            target="_blank"
            title="Book Flights"
          >
            Book Your Flights
          </a>
        </div>
      </section>
    </div>
    <div>
      <section
        class="sc-AxjAm fDEmJb"
      >
        <div
          class="sc-AxheI dVkpdt"
        >
          Berlin
          Berlin
          Berlin
          Berlin
          InDirect
          213
          <a
            aria-label="Book Flights"
            class="sc-AxmLO dTodBP"
            href="https://www.skyscanner.net/transport/flights/Berl/Berl/2020-06-25/2020-06-27"
            rel="noopener noreferrer"
            target="_blank"
            title="Book Flights"
          >
            Book Your Flights
          </a>
        </div>
        <div
          class="sc-Axmtr kkOJu"
        >
          Berlin
          Germany
        </div>
        <div
          class="sc-AxheI dVkpdt"
        >
          Berlin
          Berlin
          Berlin
          Berlin
          InDirect
          192
          <a
            aria-label="Book Flights"
            class="sc-AxmLO dTodBP"
            href="https://www.skyscanner.net/transport/flights/Berl/Berl/2020-06-25/2020-06-27"
            rel="noopener noreferrer"
            target="_blank"
            title="Book Flights"
          >
            Book Your Flights
          </a>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="sc-AxjAm fDEmJb"
    >
      <div
        class="sc-AxheI dVkpdt"
      >
        Berlin
        Berlin
        Berlin
        Berlin
        InDirect
        213
        <a
          aria-label="Book Flights"
          class="sc-AxmLO dTodBP"
          href="https://www.skyscanner.net/transport/flights/Berl/Berl/2020-06-25/2020-06-27"
          rel="noopener noreferrer"
          target="_blank"
          title="Book Flights"
        >
          Book Your Flights
        </a>
      </div>
      <div
        class="sc-Axmtr kkOJu"
      >
        Berlin
        Germany
      </div>
      <div
        class="sc-AxheI dVkpdt"
      >
        Berlin
        Berlin
        Berlin
        Berlin
        InDirect
        192
        <a
          aria-label="Book Flights"
          class="sc-AxmLO dTodBP"
          href="https://www.skyscanner.net/transport/flights/Berl/Berl/2020-06-25/2020-06-27"
          rel="noopener noreferrer"
          target="_blank"
          title="Book Flights"
        >
          Book Your Flights
        </a>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
